packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -85,15 +101,73 @@ const defaultTreeConfigurationOptions: Required<ITreeConfigurationOptions> = {
 	enableSchemaValidation: false,
 };
 
+/**
+ * Property-bag configuration for {@link TreeViewConfiguration} construction.
+ * @public
+ */
+export interface ITreeViewConfiguration<TSchema extends ImplicitFieldSchema = ImplicitFieldSchema> {
+	/**
+	 * The schema which the application wants to view the tree with.
+	 */
+	readonly schema: TSchema;
+
+	/**
+	 * If `true`, the tree will validate new content against its stored schema at insertion time
+	 * and throw an error if the new content doesn't match the expected schema.
+	 *
+	 * @defaultValue `false`.
+	 *
+	 * @remarks Enabling schema validation has a performance penalty when inserting new content into the tree because
+	 * additional checks are done. Enable this option only in scenarios where you are ok with that operation being a
+	 * bit slower.
+	 */
+	readonly enableSchemaValidation?: boolean;
+}
+
+/**
+ * Configuration for {@link ITree.viewWith}.
+ * @public
+ */
+export class TreeViewConfiguration<TSchema extends ImplicitFieldSchema = ImplicitFieldSchema>
```

Comment:
Not directly related to this PR, but one pattern I think we should encourage apps to take, is to is to have a snapshot test which saves/diffs the default stored schema generated from their view schema to help them discover changes which impact schema evolution. I created `encodeTreeSchema` to facilitate this, but so much has change since then that its no longer usable for its intended purpose.

Maybe a method on TreeViewConfiguration, something like `ecodeSchema` could serve as a more discoverable and modern replacement?